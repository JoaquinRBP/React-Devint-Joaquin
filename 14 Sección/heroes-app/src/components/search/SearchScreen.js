import React, { useMemo } from 'react'
import {HeroCard} from '../heroes/HeroCard';
import {useForm} from '../../hooks/useForm';
import {useNavigate, useSearchParams} from 'react-router-dom';
import { getHeroesbyName } from '../../selectors/getHeroesbyName';
export const SearchScreen = () => {
    const navigate = useNavigate();
    const [searchParams] = useSearchParams();
    let defecto = '';
    if(searchParams.get('q')){
        defecto = searchParams.get('q')
    }
    const [values, handleInputChange] = useForm({
        searchText: defecto
    });
    const {searchText} = values;
    const heroesfiltered = useMemo(() => getHeroesbyName(defecto), [defecto]);

    const handleSearch =(e)=>{
        e.preventDefault();
        localStorage.setItem('query',searchText);
        navigate(`?q=${searchText}`)
    }
return (
    <>
        <h1>SearchScreen</h1>
        <hr />
        <div className="row">
            <div className="col-5">
                <h4>Seacrh Form</h4>
                <hr />
                <form onSubmit={handleSearch}>
                    <input 
                    type="text" 
                    className="form-control" 
                    placeholder="Find your hero"
                    name="searchText"
                    value={searchText}
                    onChange={handleInputChange}
                    />
                    <button
                        className="btn btn-primary"
                        type="submit"
                        >
                            Search...
                        </button>
                </form>
            </div>
            <div className="col-7">
                <h4>Result</h4>
                { (defecto==='') && <div className="alert alert-info">
                    Search a Hero
                </div>}
                { (defecto!=='' && heroesfiltered.length===0) && <div className="alert alert-danger">
                    There is no a Hero with {defecto} in the list
                </div>}
                    {
                        heroesfiltered.map( hero =>(
                            <HeroCard 
                                key={hero.id}
                                {...hero}
                            />
                        ))
                    }
            </div>
        </div>
    </>
)
}
