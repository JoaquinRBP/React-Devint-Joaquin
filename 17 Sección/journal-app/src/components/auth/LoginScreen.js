import React from 'react'
import { Link } from 'react-router-dom'
import {useForm} from '../../hooks/useForm';
import {useDispatch} from 'react-redux';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { EvalError, removeError } from '../../actions/ui';
import { useSelector } from 'react-redux';
export const LoginScreen = () => {
    //useDispatch -- hook
    const dispatch =useDispatch();
    const [values, handleInputChange] = useForm({
        email:'joaquin@gmail.com',
        password:123456
    });
    const {loading} = useSelector( state => state.ui);
    const {email, password} = values;
    const handleLogin = (e) => {
        e.preventDefault();
        if(isFormValid()){
            dispatch(startLoginEmailPassword(email,password));
        }
    }
    const handleGoogleLogin = (e) => {
        e.preventDefault();
        dispatch(startGoogleLogin());
    }
    const isFormValid = () => {
        if(!validator.isEmail(email)){
            dispatch(EvalError('Email is not valid'));
            //console.log('');
            return false;
        }else if(password.length<5){
            dispatch(EvalError('Password is not valid'));
            //console.log('Password is not valid');
            return false;
        }
            dispatch(removeError());
        return true;
    }
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form 
        className="animate__animated animate__fadeIn animate__faster"
        onSubmit={handleLogin}>
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
                value={email}
                onChange={handleInputChange}
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
                value={password}
                onChange={handleInputChange}
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block"
                disabled={loading}
            >
                Login
            </button>
            <div className="auth__social-networks">
                <p>Login with Social Networks</p>
                    <div 
                        className="google-btn"
                        onClick={handleGoogleLogin}
                    >
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                    </div>
            </div>
            <Link className="link" to="/auth/register">
                Create a new account
            </Link>
        </form>
    </>
)
}
