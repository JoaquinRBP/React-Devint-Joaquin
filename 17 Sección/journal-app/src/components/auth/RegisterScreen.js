import React from 'react'
import { Link } from 'react-router-dom'

export const RegisterScreen = () => {
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form>
            <input 
                type="text"
                placeholder="Name"
                name="name"
                className="auth__input"
            />
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
            />
            <input 
                type="pasword"
                placeholder="Confirm"
                name="password2"
                className="auth__input"
                autoComplete="off"
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block mb-5"
            >
                Register
            </button>
            <Link className="link mt-5" to="/auth/login">
                All Ready Register?
            </Link>
        </form>
    </>
)
}
