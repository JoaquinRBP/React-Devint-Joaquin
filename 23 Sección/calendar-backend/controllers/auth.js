const {response} = require('express');
const { validationResult} = require('express-validator');
const crearUsuario = (req,res = response) => {
    
    const {nombre, email, password} = req.body;
    // if(nombre.length <5) {
    //     return res.status(400).json({
    //         ok: false,
    //         msg:'El nombre debe de ser mayor o igual a 5 letras'
    //     })
    // }

    res.status(201).json(
        {
            ok: true,
            msg:'register',
            nombre,
            email,
            password
        }
        ); 
}
const loginUsuario = (req,res=response) => {
    const {email, password} = req.body
    
    
    res.json({
        ok: true,
        msg:'Login',
        email,
        password
    });
}
const revalidarToken = (req,res=response) => {
    res.json({
        ok: true,
        msg: 'renew'
    });
}
module.exports = {
    crearUsuario,
    loginUsuario,
    revalidadToken: revalidarToken,
};