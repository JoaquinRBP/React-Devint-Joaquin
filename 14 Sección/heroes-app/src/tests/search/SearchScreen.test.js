import React from 'react';
import {mount} from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router';
import {SearchScreen} from '../../components/search/SearchScreen';
import {AuthContext} from '../../auth/AuthContext';
jest.mock('react-router',() => ({
    ...jest.requireActual('react-router'),
    useNavigate: ()=> jest.fn()
}));
describe('Pruebas en Search', () => {
    test('Debe de mostrarse correctamente', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Routes>
                <Route path="/search" element={<SearchScreen />}/>
                </Routes>
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.alert-info').text().trim()).toBe('Search a Hero')
    });
    test('Debe de mostrar a batman', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Routes>
                <Route path="/search" element={<SearchScreen />}/>
                </Routes>
            </MemoryRouter>
        );
        expect(wrapper.find('input').prop('value')).toBe('batman');
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar un error si no hay hero', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <Routes>
                <Route path="/search" element={<SearchScreen />}/>
                </Routes>
            </MemoryRouter>
        );
        expect(wrapper.find('.alert-danger').text().trim()).toBe('There is no a Hero with batman123 in the list');
    });
    test('Simular un cambio en input', () => {
        //Codigo distinto ya que se us√≥ useNavigate,useLocation y useParams en el proyecto en ves de history
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <Routes>
                <Route path="/search" element={<SearchScreen />}/>
                </Routes>
            </MemoryRouter>
        );
        wrapper.find('input').simulate('change',{
            target:{
                name: 'SearchText',
                value: 'batman'
            }
        })
        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
