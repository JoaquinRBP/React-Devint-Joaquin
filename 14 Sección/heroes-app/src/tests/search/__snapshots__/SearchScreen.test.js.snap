// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Search Debe de mostrar a batman 1`] = `
Array [
  <h1>
    SearchScreen
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Seacrh Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn btn-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="car-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card- title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
        Batman
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en Search Debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    SearchScreen
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Seacrh Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <div
        className="alert alert-info"
      >
        Search a Hero
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en Search Simular un cambio en input 1`] = `
Array [
  <h1>
    SearchScreen
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Seacrh Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman123"
        />
        <button
          className="btn btn-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <div
        className="alert alert-danger"
      >
        There is no a Hero with 
        batman123
         in the list
      </div>
    </div>
  </div>,
]
`;
